### ISSUE TITLE
[FEAT] AuthProvider as part of refine with integrations

### ISSUE THREAD

**Issue by hirenf14** [refinedev/refine#5159]
### Is your feature request related to a problem? Please describe.

I looked into `AuthProvider` examples. All of them are exemplified into refine's `AuthProvider` interface usage.
I think it would be of great help to have it implemented as part of `refine` itself, similar to how `DataProvider` is within the repo.

### Describe alternatives you've considered

As an alternative solution, 
```js
import { useAuthProvider } from "@refinedev/auth0";

const authProvider = useAuthProvider({
      clientId: `AUTH0_CLIENT_ID`,
      domain: `AUTH0_DOMAIN`,
});
```

### Describe the thing to improve

I looked into how DataProvider's ease the integration, similar to that `AuthProvider` can be integrated into the repo.
Referring to `next-auth`, I would like to suggest integration of `AuthProvider` like `@refinedev/auth0/react`.

**Comment by omeraplak** [comment]
That sounds good, would you like to create a refine package for Auth0?

**Comment by hirenf14** [comment]
> That sounds good, would you like to create a refine package for Auth0?

Yes, I am looking for the same. I have started for [Kinde](https://kinde.com/) actually.

I hope it works fine, once it does I will move it further.

**Comment by omeraplak** [comment]
Great! We'd like to gift you a Refine Swag Kit for your efforts and interest. We'll send you with a gift code you can use at https://store.refine.dev! Thank you.

**Comment by hirenf14** [comment]
Here's the first installment of `AuthProvider`, It's for [Kinde](https://kinde.com) with react.
https://github.com/hirenf14/refine-auth-kinde-react
